### PR TITLE
feat: New Landing Swiper

### PR DESCRIPTION
**Pull request by conrad-vanl**

### CODE REVIEW

**Comment by conrad-vanl** [comment]
Current progress

https://user-images.githubusercontent.com/103927/150806389-49f7e071-d5b1-4619-8a69-4bf3c864e4ca.mp4

https://user-images.githubusercontent.com/103927/150806500-bef599a7-bc95-446e-a1ab-4157bc33049a.mp4

**Comment by conrad-vanl** [comment]
@vinnyjth I'm struggling to understand how to get tests passing. The snapshots in any UI package are failing with this error (some snaps succeed though, oddly enough):

Screen Shot 2022-01-24 at 8 57 09 AM
.

I have tried upgrading babel versions, babel-jest, jest, etc. I've tried adding any offending package I can find to the jest ignore list. Didn't help. I've tried converting our code base to [using native ESM](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) which I think might have a shot at fixing this, but a) I don't know why it would be suddenly required based on what I see in the yarn.lock, and b) it could be a larger effort change. Ideas?

**Comment by vinnyjth** [comment]
> @vinnyjth I'm struggling to understand how to get tests passing. The snapshots in any UI package are failing with this error (some snaps succeed though, oddly enough):
> 
> I have tried upgrading babel versions, babel-jest, jest, etc. I've tried adding any offending package I can find to the jest ignore list. Didn't help. I've tried converting our code base to [using native ESM](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) which I think might have a shot at fixing this, but a) I don't know why it would be suddenly required based on what I see in the yarn.lock, and b) it could be a larger effort change. Ideas?

I haven't seen this before (at least not recently). I'll pull down and see if I can reproduce sometime soon.

**Comment by vinnyjth** [comment]
At a glance, I'm assuming it's the addition of the `react-navigation-shared-element` package. Maybe try adding a jest mock for that whole module?

**Comment by conrad-vanl** [comment]
Also @vinnyjth tests are failing in the `ui-kit` package just when I run `yarn test` in that directory, but `react-navigation-shared-element` isn't a dependency of that package (just `ui-onboarding`). do you still think it's that package?

**Comment by vinnyjth** [comment]
Probably not, in that case :/ Could it be d3? Seems very unlikely, but that error seems reminiscent of errors caused by adding modules that jest struggles to handle (and you just need to import that module _somewhere_ for everything to fall apart. .

**Comment by conrad-vanl** [comment]
If I search my `node_modules` folder for `import _extends from "@babel/runtime/helpers/extends";` the only hits are:
Screen Shot 2022-01-24 at 11 10 39 AM
:

FWIW, I tried adding this entire package to the jest ignore list...but didn't help. I also don't think our code (or its dependencies) require this package directly, I almost feel like it's the package Jest uses for code syntax highlighting, and for some reason it's grabbing the ESM version of the package now where it wasn't before.

**Comment by conrad-vanl** [comment]
https://github.com/facebook/jest/issues/12120

**Comment by conrad-vanl** [comment]
@vinnyjth this is "done" from a storybook perspective...got to write tests, make sure the API footprint is right, and hook up to the main app -
https://user-images.githubusercontent.com/103927/151108258-4a756c21-3586-464c-b4c5-0247ad05383e.mp4

Would love your thoughts on overall approach before I do that. Essentially I am changing the API for how we structure onboarding slides, moving from the old swiper component to using react-navigation.

Files to look at:

- [packages/apollos-ui-onboarding/src/LandingSwiper/index.js](https://github.com/ApollosProject/apollos-apps/pull/2338/files#diff-995d38b4f77cb01ac0a98fb07313c1dcd750826479c3f16a075b6effabe9864e): New LandingSwiper entry point: takes in a `slides` prop that converts to react-navigation routes
- [packages/apollos-ui-onboarding/src/LandingSwiper/LandingSlide.js](https://github.com/ApollosProject/apollos-apps/pull/2338/files#diff-04621eb88ab74828b98d124318d996070ba3e80ac8a56c7fc51e55fa6b006b14): A reusable "Slide" component that sets up the phone background and renders `SharedElement`s that are used to automatically create the fancy transitions from slide to slide
- For examples of what a LandingSlide looks like, checkout [packages/apollos-ui-onboarding/src/LandingSwiper/slides.js](https://github.com/ApollosProject/apollos-apps/pull/2338/files#diff-6a930cb47aaa73504263237ce7f99b21ab587978e9f45886d664e0fceec1779d)

----

I also created a refactor for `AvatarCloud` that matches the component we have in Figma, and will be used for Groups feature set. It mimics iOS Group Messaging avatars and uses a circle-fitting algorithm with D3. We might want to split this component refactor into its own PR, which I'm happy to do.

**Comment by conrad-vanl** [comment]
Dark mode

https://user-images.githubusercontent.com/103927/151109692-5550c698-dc89-4c8c-96fa-a4a44d555684.MOV

**Comment by vinnyjth** [comment]
@conrad-vanl This PR looks fantastic. 

I really like the new API for the Swiper component. I wasn't able to fully Grok the underlying components at a glance, but the external facing slide API looks incredible.

I see nothing wrong with including the revised AvatarCloud in this PR :) 

Let me know when you're ready to review this PR.

**Comment by conrad-vanl** [comment]
@vinnyjth ready, I hope

File: packages/apollos-ui-onboarding/package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     ],
     "modulePathIgnorePatterns": [
       "node_modules/@storybook/"
-    ],
+    ],
```
Comment (vinnyjth) [inline packages/apollos-ui-onboarding/package.json:42]:
```suggestion
    ],
```

**Comment by conrad-vanl** [comment]
@vinnyjth I'm happy to get the tests and such passing that was me being lazy and not fully checking it

**Review by vinnyjth — APPROVED** [review]
This PR looks great! I LOVE the new landing screen in the app. Looks great and feels super smooth. A real level up in presentation.

**Comment by vinnyjth** [comment]
@conrad-vanl do you want me to take a stab at fixing the tests?

**Comment by conrad-vanl** [comment]
Oh, hai, I never went back and fixed the tests

**Comment by conrad-vanl** [comment]
Looks like just snapshot updates maybe?

**Comment by vinnyjth** [comment]
@conrad-vanl I'm seeing this in both Github and locally

image

**Comment by vinnyjth** [comment]
@conrad-vanl I added a new mock which fixed things. Take a peek and make sure sure you're okay with it: https://github.com/ApollosProject/apollos-apps/pull/2338/commits/874cef386b9c39f0986e0803d585aae64861cdb6

**Comment by vinnyjth** [comment]
@conrad-vanl merge conflicts now 😬

**Comment by conrad-vanl** [comment]
yeah will do

**Comment by conrad-vanl** [comment]
@vinnyjth I fixed the merge conflicts but not able to test yet (M1 issues I think or just need to clear out build caches after working on other PRs like the RN upgrade...)

**Comment by vinnyjth** [comment]
Tests are failing again. I'll try and fix.

**Comment by vinnyjth** [comment]
@conrad-vanl I'm getting a crash when adding a prayer. Here's a screenshot of two of the errors from the crash, and I'll upload a gif as well. The crash does not occur on `next`. I do _not_ have an avatar, and there aren't any other prayers currently "on deck". 

Screen Shot 2022-02-03 at 10 03 44 AM
Screen Shot 2022-02-03 at 10 03 57 AM

**Comment by vinnyjth** [comment]
Here's a gif. The crash changed in location slightly, now it crashes when _closing_ the prayer screen, and I get a crash when refreshing the app.

**Comment by conrad-vanl** [comment]
@vinnyjth I'm not sure _why_ this is happening, but see if the commit I just pushed fixes for you.

**Pull request merged**